bot/seasons/evergreen/battleship.py

```diff
@@ -107,211 +111,227 @@ def __init__(
-                except asyncio.TimeoutError:
-                    await self.turn.send("You took too long. Game over!")
-                    await self.next.send(f"{self.turn} took too long. Game over!")
-                    self.gameover = True
-                    break
-                else:
-                    square = self.get_square(self.grids[self.next], self.match.string)
-                    if square.aimed:
-                        await self.turn.send("You've already aimed at this square!", delete_after=3.0)
-                    else:
-                        break
-
             if self.gameover:
-                await self.game_over(timeout=True)
-                break
+                return
 
+            square = await self.take_turn()
             square.aimed = True
-            await turn_message.delete()
+
             for message in alert_messages:
                 await message.delete()
 
             alert_messages = []
-            alert_messages.append(await self.next.send(f"{self.turn} aimed at {self.match.string}!"))
+            alert_messages.append(await self.next.user.send(f"{self.turn.user} aimed at {self.match.string}!"))
 
             if square.boat:
-                await self.turn.send("Hit!", delete_after=3.0)
-                alert_messages.append(await self.next.send("Hit!"))
-                if self.check_sink(self.grids[self.next], square.boat):
-                    await self.turn.send(f"You've sunk their {square.boat} ship!", delete_after=3.0)
-                    alert_messages.append(await self.next.send(f"Oh no! Your {square.boat} ship sunk!"))
-                    if self.check_gameover(self.grids[self.next]):
-                        await self.turn.send("You win!")
-                        await self.next.send("You lose!")
-                        self.gameover = True
-                        await self.game_over(winner=self.turn, loser=self.next)
-                        break
+                await self.hit(square, alert_messages)
+                if self.gameover:
+                    return
             else:
-                await self.turn.send("Miss!", delete_after=3.0)
-                alert_messages.append(await self.next.send("Miss!"))
+                await self.turn.user.send("Miss!", delete_after=3.0)
+                alert_messages.append(await self.next.user.send("Miss!"))
 
             self.turn, self.next = self.next, self.turn
 
 
 class Battleship(commands.Cog):
-    """Play the classic game Battleships!"""
+    """Play the classic game Battleship!"""
 
     def __init__(self, bot: commands.Bot) -> None:
         self.bot = bot
         self.games: typing.List[Game] = []
         self.waiting: typing.List[discord.Member] = []
 
+    def predicate(
+        self,
+        ctx: commands.Context,
+        announcement: discord.Message,
+        reaction: discord.Reaction,
+        user: discord.Member
+    ) -> typing.Optional[bool]:
```

Comment:
I don't see where this can return `None`.

```suggestion
    ) -> bool:
```